packages/fx-core/src/common/wrappedAxiosClient.ts

```diff
@@ -0,0 +1,289 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+import axios, { AxiosInstance, CreateAxiosDefaults } from "axios";
+import { TOOLS } from "../core/globalVars";
+import { APP_STUDIO_API_NAMES } from "../component/driver/teamsApp/constants";
+import {
+  TelemetryPropertyKey,
+  TelemetryPropertyValue,
+} from "../component/driver/teamsApp/utils/telemetry";
+import { TelemetryEvent, TelemetryProperty } from "./telemetry";
+import { DeveloperPortalAPIFailedError } from "../error/teamsApp";
+import { Constants } from "../component/driver/teamsApp/constants";
+import { HttpMethod } from "../component/constant/commonConstant";
+
+/**
+ * This client will send telemetries to record API request trace
+ */
+export class WrappedAxiosClient {
+  public static create(config?: CreateAxiosDefaults): AxiosInstance {
+    const instance = axios.create(config);
+
+    // Send API start telemetry
+    instance.interceptors.request.use((request) => {
+      const method = request.method!;
+      const fullPath = `${request.baseURL ?? ""}${request.url ?? ""}`;
+      const apiName = this.convertUrlToApiName(fullPath, method);
+
+      const properties: { [key: string]: string } = {
+        url: `<${apiName}-url>`,
+        method: method,
+        params: this.generateParameters(request.params),
+        ...this.generateExtraProperties(fullPath, request.data),
+      };
+
+      let eventName: string;
+      if (this.isTDPApi(fullPath)) {
+        eventName = TelemetryEvent.AppStudioApi;
+      } else {
+        eventName = TelemetryEvent.DependencyApi;
+      }
+
+      TOOLS.telemetryReporter?.sendTelemetryEvent(`${eventName}-start`, properties);
+
+      return request;
+    });
+
+    instance.interceptors.response.use(
+      // Send API success telemetry
+      (response) => {
+        const method = response.request.method;
+        const fullPath = `${(response.request.host as string) ?? ""}${
+          (response.request.path as string) ?? ""
+        }`;
+        const apiName = this.convertUrlToApiName(fullPath, method);
+
+        const properties: { [key: string]: string } = {
+          url: `<${apiName}-url>`,
+          method: method,
+          params: this.generateParameters(response.config.params),
+          [TelemetryPropertyKey.success]: TelemetryPropertyValue.success,
+          "status-code": response.status.toString(),
+          ...this.generateExtraProperties(fullPath, response.data),
+        };
+
+        let eventName: string;
+        if (this.isTDPApi(fullPath)) {
+          eventName = TelemetryEvent.AppStudioApi;
+        } else {
+          eventName = TelemetryEvent.DependencyApi;
+        }
+        TOOLS.telemetryReporter?.sendTelemetryErrorEvent(eventName, properties);
+        return response;
+      },
+      // Send API failure telemetry
+      (error) => {
```

Comment:
For 4xx/5xx response, does it go to `(response) =>` or `(error) =>` ?

Reply:
4xx and 5xx goes to error